### ISSUE TITLE
Spurious cursor06 failure

### ISSUE THREAD

**Issue by sueloverso** [wiredtiger/wiredtiger#1813]
In my pull request I saw this failure from 
http://build.wiredtiger.com:8080/job/wiredtiger-pull-request-tester/3184/

```
======================================================================
ERROR: test_cursor06.test_cursor06.test_reconfigure_overwrite(table-S) (subunit.RemotedTestCase)
test_cursor06.test_cursor06.test_reconfigure_overwrite(table-S)
----------------------------------------------------------------------
_StringException: Traceback (most recent call last):
  File "/home/jenkins/jenkins/workspace/wiredtiger-pull-request-tester/test/suite/test_cursor06.py", line 65, in test_reconfigure_overwrite
    self.session.drop(uri, "force")
WiredTigerError: No such file or directory
```
I have not been able to reproduce this failure manually.  When I went to tinderbox, where it ran, the `WT_TEST` directory did not save a copy of the failing test's directory even though the results.txt file in there indicated it would be preserved.

Telling autobuild to retest and rerunning the pull-request-tester Jenkins job passed with no changes.  There is little to go on here, but in case anyone else has seen a spurious ENOENT.

**Comment by agorrod** [comment]
@sueloverso That is interesting. I first saw the failure testing a branch yesterday (http://build.wiredtiger.com:8080/job/wiredtiger-pull-request-tester/3177/). I merged that branch and started seeing the failure on develop: http://build.wiredtiger.com:8080/job/wiredtiger-test-unit/3869/

The change from yesterday (#1808) is reverted on develop now. I also couldn't reproduce the failure locally.

tl;dr the failure is something that's in current develop not your branch (or the one I reverted yesterday).

**Comment by sueloverso** [comment]
FTR the output is very similar (ENOENT) but slightly different output in the 3869 failure.  It is still cursor06, but it doesn't mention it is part of drop:
```
====================================================================
ERROR: test_cursor06.test_cursor06.test_reconfigure_overwrite(table-S) (subunit.RemotedTestCase)
test_cursor06.test_cursor06.test_reconfigure_overwrite(table-S)
----------------------------------------------------------------------
_StringException: Traceback (most recent call last):
  File "/home/jenkins/jenkins/workspace/wiredtiger-test-unit/test/suite/test_cursor06.py", line 66, in test_reconfigure_overwrite
    self.pop(uri)
  File "/home/jenkins/jenkins/workspace/wiredtiger-test-unit/test/suite/test_cursor06.py", line 50, in pop
    complex_populate(self, uri, 'key_format=' + self.keyfmt, 100)
  File "/home/jenkins/jenkins/workspace/wiredtiger-test-unit/test/suite/helper.py", line 178, in complex_populate
    complex_populate_type(self, uri, config, rows, '')
  File "/home/jenkins/jenkins/workspace/wiredtiger-test-unit/test/suite/helper.py", line 202, in complex_populate_type
    indxname + ':indx4', 'columns=(column2,column4)' + ',' + type)
WiredTigerError: No such file or directory
```

**Comment by sueloverso** [comment]
The one thing in common between test-unit and pull-request-tester is that they run the test suite in parallel, whereas all my local testing is serial or just the individual test.

**Comment by agorrod** [comment]
Turns out this is due to ```test_cursor06``` having scenarios with the same name. This means that when tests are run in parallel two test cases can be assigned the same directory - which leads to badness.

**Comment by sueloverso** [comment]
With @michaelcahill and @agorrod we figured out that when `scenarios` are specified, all of them must have a unique name.  In `cursor06` we have:
```
    scenarios = [
        ('file-r', dict(type='file:',keyfmt='r',complex=0)),
        ('file-S', dict(type='file:',keyfmt='S',complex=0)),
        ('lsm-S', dict(type='lsm:',keyfmt='S',complex=0)),
        ('table-r', dict(type='table:',keyfmt='r',complex=0)),
        ('table-S', dict(type='table:',keyfmt='S',complex=0)),
        ('table-r', dict(type='table:',keyfmt='r',complex=1)),
        ('table-S', dict(type='table:',keyfmt='S',complex=1)),
    ]
```
There are two `table-r` and `table-S` and when run in parallel they can conflict, resulting in directories being removed out from under threads.  It is easy to repro with `run.py -p -j 20 cursor06`.  We should check all tests that specify scenarios for this issue ideally with a script that checks.

**Comment by ddanderson** [comment]
I bet we can check this at runtime in the test harness when we run the tests.  I'll look into it.
Good sleuthing!

**Comment by michaelcahill** [comment]
@ddanderson, I had the same idea -- my suggested change is in #1818.  I'd like to get something into develop to fix this, because for some reason we're seeing it a lot in testing all of a sudden.

**Comment by agorrod** [comment]
Fixed by merge of #1818.
